components/user_list.tsx

```diff
@@ -49,16 +69,18 @@ export default class UserList extends React.PureComponent {
         let content;
         if (users == null) {
             return <LoadingScreen/>;
-        } else if (users.length > 0) {
-            content = users.map((user, index) => {
+        } else if (users.length > 0 && RowComponentType && this.props.actionProps) {
+            content = users.map((user: UserProfile, index: number) => {
+                const {actionUserProps, extraInfo} = this.props;
+                const userId = user.id;
                 return (
                     <RowComponentType
                         key={user.id}
                         user={user}
-                        extraInfo={this.props.extraInfo[user.id]}
+                        extraInfo={extraInfo ? extraInfo[userId] : {}}
```

Comment:
passing `{}` will throw error in new ts versions. we may as well pass undefined
```suggestion
                        extraInfo={extraInfo?.[userId]}
```